### PR TITLE
[Merged by Bors] - chore(Algebra/Category/MonCat): improve definitional properties in restrict scalars

### PR DESCRIPTION
**Pull request by chrisflav**
Replaces two usages of `subst` by an explicit construction in `ChangeOfRings`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by joelriou** [comment]
It would be nice to deduce these definitions from something like `AddEquiv.toLinearEquiv` (taking as input an `AddEquiv` (obtained here as `by rfl`) and the linearity condition).

**Comment by chrisflav** [comment]
> It would be nice to deduce these definitions from something like `AddEquiv.toLinearEquiv` (taking as input an `AddEquiv` (obtained here as `by rfl`) and the linearity condition).

Thanks for the suggestion, its is way cleaner now. Also that led me into the right direction for fixing the `Id` analog.

File: Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean [hunk 1]
```diff
@@ -153,19 +153,26 @@ section
 
 variable {R : Type u₁} [Ring R] (f : R →+* R) (hf : f = RingHom.id R)
 
-/-- The restriction of scalars by a ring morphism that is the identity identify to the
-identity functor. -/
-def restrictScalarsId' : ModuleCat.restrictScalars.{v} f ≅ 𝟭 _ := by subst hf; rfl
+/-- For a `R`-module `M`, the restriction of scalars of `M` by the identity morphisms identifies
+to `M`. -/
+def restrictScalarsId'_app (M : ModuleCat R) : (restrictScalars f).obj M ≅ M :=
```
Comment (joelriou) [inline Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean:158]:
```suggestion
def restrictScalarsId'App (M : ModuleCat R) : (restrictScalars f).obj M ≅ M :=
```

File: Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean [hunk 2]
```diff
@@ -180,21 +187,27 @@ section
 variable {R₁ : Type u₁} {R₂ : Type u₂} {R₃ : Type u₃} [Ring R₁] [Ring R₂] [Ring R₃]
   (f : R₁ →+* R₂) (g : R₂ →+* R₃) (gf : R₁ →+* R₃) (hgf : gf = g.comp f)
 
+/-- For each `R₃`-module `M`, restriction of scalars of `M` by a composition of ring morphisms
+identifies to successively restricting scalars. -/
+def restrictScalarsComp'_app (M : ModuleCat R₃) :
```
Comment (joelriou) [inline Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean:192]:
```suggestion
def restrictScalarsComp'App (M : ModuleCat R₃) :
```

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**